src/Interpreters/Context.cpp

```diff
@@ -3430,7 +3430,8 @@ void Context::initializeTraceCollector()
 /// Call after unexpected crash happen.
 void Context::handleCrash() const TSA_NO_THREAD_SAFETY_ANALYSIS
 {
-    shared->system_logs->handleCrash();
+    if (shared->system_logs)
```

Comment:
Since it's possible that `shared->system_logs` is being initialized we can have a race here, that's why `TSA_NO_THREAD_SAFETY_ANALYSIS` are used, am I right? But I suppose we cannot get segfault, because `shared->system_logs` is only set once from nullptr to value during initialization.

Reply:
> Since it's possible that shared->system_logs is being initialized we can have a race here, that's why TSA_NO_THREAD_SAFETY_ANALYSIS are used, am I right?

Yes, but we never initialize it for Keeper so it should be okay